sdk/identity/identity/src/credentials/visualStudioCodeCredential.ts

```diff
@@ -145,18 +129,69 @@ export class VisualStudioCodeCredential implements TokenCredential {
     checkUnsupportedTenant(this.tenantId);
   }
 
+  private async loadBrokerPlugin() {
+    try {
+      // @ts-ignore
+      const { nativeBrokerPlugin } = await import("@azure/identity-broker");
+      return nativeBrokerPlugin;
+    } catch (error) {
+      // Package not installed, return null or fallback
+      logger.info("The @azure/identity-broker package is not installed.");
+      return undefined;
+    }
+  }
+
+  private loadVSCodeAuthRecord(): AuthenticationRecord | undefined {
+    const authRecordPath = path.join(os.homedir(), ".azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+
+    if (!fs.existsSync(authRecordPath)) {
+      logger.info(`The Visual Studio Code authentication record file does not exist.`);
+      return undefined;
+    }
+
+    const authRecordContent = fs.readFileSync(authRecordPath, { encoding: 'utf8' });
+    return deserializeAuthenticationRecord(authRecordContent);
+  }
+
   /**
    * Runs preparations for any further getToken request.
    */
   private async prepare(): Promise<void> {
-    // Attempts to load the tenant from the VSCode configuration file.
-    const settingsTenant = getPropertyFromVSCode("azure.tenant");
-    if (settingsTenant) {
-      this.tenantId = settingsTenant;
+    // check if broker is available
+    // check and parse authentication record
+    const tenantId =
+      processMultiTenantRequest(
+        this.tenantId,
+        this.options,
+        this.additionallyAllowedTenantIds,
+        logger,
+      ) || this.tenantId;
+
+    const nativeBrokerPlugin = await this.loadBrokerPlugin();
+    const authenticationRecord = this.loadVSCodeAuthRecord();
+
+    if (!nativeBrokerPlugin || !authenticationRecord) {
+      throw new CredentialUnavailableError("Visual Studio Code Authentication is not available."
```

Comment:
later on y'all might add a troubleshooting section and link to it from here like we do in other errors in identity